samples/showcase.cpp

```diff
@@ -0,0 +1,220 @@
+    int opt;
+    int option_index = 0;
+    while ((opt = getopt_long(argc, argv, OPTSTR, OPTIONS, &option_index)) >= 0) {
+        std::string arg(optarg ? optarg : "");
+        switch (opt) {
+            default:
+            case 'h':
+                printUsage(argv[0]);
+                exit(0);
+            case 'a':
+                if (arg == "opengl") {
+                    app->config.backend = Engine::Backend::OPENGL;
+                } else if (arg == "vulkan") {
+                    app->config.backend = Engine::Backend::VULKAN;
+                } else {
+                    std::cerr << "Unrecognized backend. Must be 'opengl'|'vulkan'." << std::endl;
+                }
+                break;
+            case 'i':
+                app->config.iblDirectory = arg;
+                break;
+            case 'p':
+                app->shadowPlane = true;
+                break;
+        }
+    }
+    return optind;
+}
+
+static std::ifstream::pos_type getFileSize(const char* filename) {
+    std::ifstream in(filename, std::ifstream::ate | std::ifstream::binary);
+    return in.tellg();
+}
+
+int main(int argc, char** argv) {
+    App app;
+
+    app.config.title = "showcase";
+    app.config.iblDirectory = FilamentApp::getRootPath() + DEFAULT_IBL;
+
+    int option_index = handleCommandLineArgments(argc, argv, &app);
+    utils::Path filename;
+    int num_args = argc - option_index;
+    if (num_args >= 1) {
+        filename = argv[option_index];
+        if (!filename.exists()) {
+            std::cerr << "file " << argv[option_index] << " not found!" << std::endl;
+            return 1;
+        }
+    }
+
+    auto setup = [&app, filename](Engine* engine, View* view, Scene* scene) {
+        FilamentApp::get().getIBL()->getIndirectLight()->setIntensity(100000);
+        app.loader = AssetLoader::create(engine);
+        if (filename.isEmpty()) {
+            return;
+        }
+        long contentSize = static_cast<long>(getFileSize(filename.c_str()));
+        if (contentSize <= 0) {
+            std::cerr << "Unable to open " << filename << std::endl;
+            exit(1);
+        }
+
+        // Consume the glTF file.
+        std::ifstream in(filename.c_str(), std::ifstream::in);
+        std::vector<uint8_t> buffer(static_cast<unsigned long>(contentSize));
+        if (!in.read((char*) buffer.data(), contentSize)) {
+            std::cerr << "Unable to read " << filename << std::endl;
+            exit(1);
+        }
+
+        // Parse the glTF file and create Filament entities.
+        app.asset = app.loader->createAssetFromJson(buffer.data(), buffer.size());
+        if (!app.asset) {
+            std::cerr << "Unable to parse " << filename << std::endl;
+            exit(1);
+        }
+        buffer.clear();
+        buffer.shrink_to_fit();
+
+        // Compute the scale required to fit the model's bounding box into [-1, +1]
+        std::cout << "\nAsset min: " << app.asset->getBoundingBox().min << std::endl;
```

Comment:
Remove logs.
Also we should make this a utility function in a header somewhere. It seems useful to all of our samples.